chain/types/ethtypes/eth_types.go

```diff
@@ -1025,6 +1025,47 @@ type EthTraceTransaction struct {
 	TransactionPosition int     `json:"transactionPosition"`
 }
 
+type EthTraceFilterResult struct {
+	*EthTrace
+	BlockHash           EthHash `json:"blockHash"`
+	BlockNumber         int64   `json:"blockNumber"`
+	TransactionHash     EthHash `json:"transactionHash"`
+	TransactionPosition int     `json:"transactionPosition"`
+}
+
+// EthTraceFilterCriteria defines the criteria for filtering traces.
+type EthTraceFilterCriteria struct {
+	// Interpreted as an epoch (in hex) or one of "latest" for last mined block, "earliest" for first,
+	// "pending" for not yet committed messages.
+	// Optional, default: "latest".
+	FromBlock *string `json:"fromBlock,omitempty"`
+
+	// Interpreted as an epoch (in hex) or one of "latest" for last mined block, "earliest" for first,
+	// "pending" for not yet committed messages.
+	// Optional, default: "latest".
+	ToBlock *string `json:"toBlock,omitempty"`
+
+	// Actor address or a list of addresses from which event logs should originate.
+	// Optional, default: nil.
+	// The JSON decoding must treat a string as equivalent to an array with one value, for example
+	// "0x8888f1f195afa192cfee86069858" must be decoded as [ "0x8888f1f195afa192cfee86069858" ]
+	FromAddress *[]string `json:"fromAddress,omitempty"`
```

Comment:
these two slices don't need to be pointers, I don't think you gain anything from having an additional layer of `nil` and needing to dereference these. But, there is a `EthAddressList` type you could use instead of `[]string` here, which does the special single vs array json decoding for you, so you should probably swap them out for that. See `EthFilterSpec` and `EthSubscriptionParams` which have the same comment about decoding.

Reply:
347383772a91a0a0114b3dbee60f8edeb9f0bb84 and 7598de6fc2c3c0f26aa15a7a800c936ad6a8028a